bakerytest/bakerytest_test.go

```diff
@@ -142,7 +145,11 @@ func (s *suite) TestInteractiveDischarger(c *gc.C) {
 	var d *bakerytest.InteractiveDischarger
 	d = bakerytest.NewInteractiveDischarger(nil, http.HandlerFunc(
 		func(w http.ResponseWriter, r *http.Request) {
-			d.FinishInteraction(w, r, nil)
+			d.FinishInteraction(w, r, []checkers.Caveat{
+				checkers.Caveat{
+					Condition: "test pass",
```

Comment:
Rather than overloading failChecker, whose sole purpose is to fail when checking caveats, I'd suggest using a custom checker below that only succeeds when it sees the condition specified here.